src/main/scala/chisel3/util/experimental/decode/TruthTable.scala

```diff
@@ -29,40 +30,107 @@ sealed class TruthTable private (val table: Seq[(BitPat, BitPat)], val default:
 
 object TruthTable {
 
-  /** Convert a table and default output into a [[TruthTable]]. */
-  def apply(table: Iterable[(BitPat, BitPat)], default: BitPat, sort: Boolean = true): TruthTable = {
+  /** Pad the input signals to equalize all input widths. Pads input signals
+    *  to the maximum width found in the table.
+    *
+    * @param table the truth table whose rows will be padded
+    * @return the same truth table but with inputs padded
+    */
+  private def padInputs(table: Iterable[(BitPat, BitPat)]): Iterable[(BitPat, BitPat)] = {
     val inputWidth = table.map(_._1.getWidth).max
-    require(table.map(_._2.getWidth).toSet.size == 1, "output width not equal.")
-    val outputWidth = table.map(_._2.getWidth).head
-    val mergedTable = table.map {
-      // pad input signals if necessary
+    table.map {
       case (in, out) if inputWidth > in.width =>
         (BitPat.N(inputWidth - in.width) ## in, out)
       case (in, out) => (in, out)
     }
-      .groupBy(_._1.toString)
-      .map {
-        case (key, values) =>
-          // merge same input inputs.
-          values.head._1 -> BitPat(s"b${Seq
-            .tabulate(outputWidth) { i =>
-              val outputSet = values
-                .map(_._2)
-                .map(_.rawString)
-                .map(_(i))
-                .toSet
-                .filterNot(_ == '?')
-              require(
-                outputSet.size != 2,
-                s"TruthTable conflict in :\n${values.map { case (i, o) => s"${i.rawString}->${o.rawString}" }.mkString("\n")}"
-              )
-              outputSet.headOption.getOrElse('?')
-            }
-            .mkString}")
+  }
+
+  /** For each duplicated input, combine the outputs into a single Seq.
+    *
+    * @param table the truth table
+    * @return a Seq of tuple of length 2, where the first element is the
+    *         input and the second element is a Seq of combined outputs
+    *         for the input
+    */
+  private def mergeTableOnInputs(table: Iterable[(BitPat, BitPat)]): Seq[(BitPat, Seq[BitPat])] = {
+    groupByIntoSeq(table)(_._1).map {
+      case (input, mappings) =>
+        input -> mappings.map(_._2)
+    }
+  }
+
+  /** Check for BitPats that have non-zero overlap
+    *
+    * Non-zero overlap means that for two BitPats a and b, there is at least
+    * one bitpos where the indices at the bitpos for the two bits are
+    * present in [(0, 1), (1, 0)].
+    *
+    * @param x List of BitPats to check for overlap
+    * @return true if the overlap is non-zero, false otherwise
+    */
+  private def checkDups(x: BitPat*): Boolean = {
```

Comment:
```suggestion
  private def checkForNonZeroOverlap(x: BitPat*): Boolean = {
```

? "check duplicates" doesn't sound like the right name for this function given the scaladoc description?